### PR TITLE
Frontend: implement shortcut panel

### PR DESCRIPTION
**Pull request by alrapal**
# Description:
- Creation of a component shortcut
- Creation of a component shortcut panel, handling a list of shortcuts
- Styling using tailwind only.

# Screenshots: 
Screenshot 2022-12-26 at 15 30 03

related issue: #7

### CODE REVIEW

**Review by ktpa — CHANGES_REQUESTED** [review]
I've noticed that the shortcut panel doesn't render properly when testing it, see the video below.

https://www.youtube.com/watch?v=KFrxaPpCS8o

- The shortcuts are shown below the graph options, even if they are not selected.
- The shortcuts cause, upon selection, a second vertical scroll bar and a new horizontal one to appear.

This has to be addressed before merging.

**Comment by alrapal** [comment]
> I've noticed that the shortcut panel doesn't render properly when testing it, see the video below.
> 
> https://www.youtube.com/watch?v=KFrxaPpCS8o
> 
> * The shortcuts are shown below the graph options, even if they are not selected.
> * The shortcuts cause, upon selection, a second vertical scroll bar and a new horizontal one to appear.
> 
> This has to be addressed before merging.

Nice catch,
I'll look into it, could you give me some info regarding the browser and OS?

**Comment by alrapal** [comment]
I addressed the duplicated shortcut panel, came from some code missing. But I cannot reproduce the scrollbar, could you give more details regarding this @ktpa ? (Browser and OS)

**Comment by ktpa** [comment]
> I addressed the duplicated shortcut panel, came from some code missing. But I cannot reproduce the scrollbar, could you give more details regarding this @ktpa ? (Browser and OS)

I'm on Windows 10 this time using 1440p and Chrome.

**Comment by RuthgerD** [comment]
Maybe make the settings window a bit larger? seems a bit dinky

File: frontend/src/components/shortcut/Shortcut.svelte [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    // Label attached to the input text. Desciption of the shortcut
+    export let shortcut_label: string = "Not defined";
+    // Key to be displayed within the input text
+    export let shortcut_key: string;
+
+    const dispatch: Function = createEventDispatcher();
+
+    // Dispatch an event when the value of the input has been changed. Useful for rebinding
+    const change_event = (event: Event) => {
+        dispatch('change', event)
```
Comment (RuthgerD) [inline frontend/src/components/shortcut/Shortcut.svelte:13]: event seems to be unused atm

Comment (alrapal) [reply frontend/src/components/shortcut/Shortcut.svelte:13]: Yes, it is made with the possibility to rebind in mind. But if you rather like, I can remove the code related to rebinding

**Comment by alrapal** [comment]
> Maybe make the settings window a bit larger? seems a bit dinky
I can make it larger
In height or width? =)

**Comment by alrapal** [comment]
Would that be better? 

(14' screen)
Screenshot 2023-01-02 at 14 36 07

**Comment by RuthgerD** [comment]
Ideally it would scale with screen size (up to a limit) but this is ok for now

**Comment by alrapal** [comment]
> Ideally it would scale with screen size (up to a limit) but this is ok for now

It is currently scaling, but without limit, so it should adapt to screen size
(32´screen)
Screenshot 2023-01-02 at 15 46 22

**Review by ktpa — CHANGES_REQUESTED** [review]
Horizontal scroll bar is fixed.
So is the unintentional double display.

However, this persists:
image

It should only be one scroll bar, not two.

Then to Ruthger's point, resize settings window to make settings more visible, perhaps not by making it larger from all areas but only in the vertical aspect of the panel itself. So the panel becomes "taller".

**Review by ktpa — CHANGES_REQUESTED** [review]
Looks good! But scrollbar is still there.

After some testing myself, removing  overflow-y-scroll from the div in ShortcutPanel.svelte makes the extra scrollbar go away. Is  overflow-y-scroll necessary for some other reason or is this just an oversight? It seems to work fine on my end without it.

**Review by ktpa — APPROVED** [review]
LGTM

edit: the branch cannot be rebased.. :grimacing:

**Pull request merged**